test/e2e/util/hypershift_framework.go

```
@@ -137,6 +138,21 @@ func (h *hypershiftTest) before(hostedCluster *hyperv1.HostedCluster, opts *Plat
 				ValidateAuthenticationSpec(t, h.ctx, h.client, hostedCluster, opts.ExtOIDCConfig)
 			}
 		}
+		if opts.ExternalCNIProvider == "cilium" {
+			// Only install Cilium when there are worker nodes configured.
+			// The cilium-olm deployment requires worker nodes to schedule its pods.
+			// TestNodePool sets NodePoolReplicas=0 and creates NodePools later in individual tests,
+			// so we skip Cilium installation during the initial cluster validation phase.
+			if !util.IsPrivateHC(hostedCluster) && opts.NodePoolReplicas > 0 {
```

Hmm. This condition seems to be new? 
Can you highlight what changes you did in the last "force-push"? I may have missed something. It would be better to push new commits incrementally to make it easier for the reviewer. Force-pushing to the same commit makes me review everything again :-/
Anyway, maybe it would be better to make the test fail fast if NodePoolReplicas is 0. Something like this:
```
if !util.IsPrivateHC(hostedCluster) {
   if opts.NodePoolReplicas == 0 {
      t.Fatal("NodePool replicas must be positive for Cilium to install")
   }
   guestClient := WaitForGuestClient(t, context.Background(), h.client, hostedCluster)
   ...
}
```
This would prevent the situation when we want cilium but it's not installed and we would find out only later through various tests failing.

---

@mgencur Fixed no body in committed message issue, please review again, thanks a lot.